src/epoch.rs

```diff
@@ -2226,6 +2242,218 @@ impl Epoch {
+    pub fn weekday(&self) -> Weekday {
+        // J1900 was a Monday so we just have to modulo the number of days by the number of days per week.
+        // The function call will be optimized away.
+        self.weekday_in_time_scale(TimeScale::TAI)
+    }
+
+    /// Returns weekday in UTC timescale
+    pub fn weekday_utc(&self) -> Weekday {
+        self.weekday_in_time_scale(TimeScale::UTC)
+    }
+
+    /// Returns the next weekday.
+    ///
+    /// ```
+    /// use hifitime::prelude::*;
+    ///
+    /// let epoch = Epoch::from_gregorian_utc_at_midnight(1988, 1, 2);
+    /// assert_eq!(epoch.weekday_utc(), Weekday::Saturday);
+    /// assert_eq!(epoch.next(Weekday::Sunday), Epoch::from_gregorian_utc_at_midnight(1988, 1, 3));
+    /// assert_eq!(epoch.next(Weekday::Monday), Epoch::from_gregorian_utc_at_midnight(1988, 1, 4));
+    /// assert_eq!(epoch.next(Weekday::Tuesday), Epoch::from_gregorian_utc_at_midnight(1988, 1, 5));
+    /// assert_eq!(epoch.next(Weekday::Wednesday), Epoch::from_gregorian_utc_at_midnight(1988, 1, 6));
+    /// assert_eq!(epoch.next(Weekday::Thursday), Epoch::from_gregorian_utc_at_midnight(1988, 1, 7));
+    /// assert_eq!(epoch.next(Weekday::Friday), Epoch::from_gregorian_utc_at_midnight(1988, 1, 8));
+    /// assert_eq!(epoch.next(Weekday::Saturday), Epoch::from_gregorian_utc_at_midnight(1988, 1, 9));
+    /// ```
+    pub fn next(&self, weekday: Weekday) -> Self {
+        let delta_days = self.weekday() - weekday;
+        if delta_days == Duration::ZERO {
+            *self + 7 * Unit::Day
+        } else {
+            *self + delta_days
+        }
+    }
+
+    pub fn next_weekday_at_midnight(&self, weekday: Weekday) -> Self {
+        self.next(weekday).with_hms_strict(0, 0, 0)
+    }
+
+    pub fn next_weekday_at_noon(&self, weekday: Weekday) -> Self {
+        self.next(weekday).with_hms_strict(12, 0, 0)
+    }
+
+    /// Returns the next weekday.
+    ///
+    /// ```
+    /// use hifitime::prelude::*;
+    ///
+    /// let epoch = Epoch::from_gregorian_utc_at_midnight(1988, 1, 2);
+    /// assert_eq!(epoch.previous(Weekday::Friday), Epoch::from_gregorian_utc_at_midnight(1988, 1, 1));
+    /// assert_eq!(epoch.previous(Weekday::Thursday), Epoch::from_gregorian_utc_at_midnight(1987, 12, 31));
+    /// assert_eq!(epoch.previous(Weekday::Wednesday), Epoch::from_gregorian_utc_at_midnight(1987, 12, 30));
+    /// assert_eq!(epoch.previous(Weekday::Tuesday), Epoch::from_gregorian_utc_at_midnight(1987, 12, 29));
+    /// assert_eq!(epoch.previous(Weekday::Monday), Epoch::from_gregorian_utc_at_midnight(1987, 12, 28));
+    /// assert_eq!(epoch.previous(Weekday::Sunday), Epoch::from_gregorian_utc_at_midnight(1987, 12, 27));
+    /// assert_eq!(epoch.previous(Weekday::Saturday), Epoch::from_gregorian_utc_at_midnight(1987, 12, 26));
+    /// ```
+    pub fn previous(&self, weekday: Weekday) -> Self {
+        let delta_days = weekday - self.weekday();
```

Comment:
@ChristopherRabotin,

the current implementation behaves funny for extreme values, ie., early dates into a timescale, because we might return a Sunday for example in the use case i'm interested in, that is prior `REF_EPOCH`.

I think we should constraint the values to at worst REF_EPOCH here. It is sane for a timescale to not wrap prior t(0) which is correctly defined. But it depends what you prefer. And I think, but i might be wrong, `Duration` wight negative centuries is already implemented. But also, Duration and Epoch are two different things.